akka-docs-new/src/test/java/jdocs/tutorial_3/DeviceTest.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package jdocs.tutorial_3;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.testkit.javadsl.TestKit;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.Optional;
+
+public class DeviceTest extends JUnitSuite {
+
+    static ActorSystem system;
+
+    @BeforeClass
+    public static void setup() {
+        system = ActorSystem.create();
+    }
+
+    @AfterClass
+    public static void teardown() {
+        TestKit.shutdownActorSystem(system);
+        system = null;
+    }
+
+    //#device-registration-tests
+    @Test
+    public void testReplyToRegistrationRequests() {
+        TestKit probe = new TestKit(system);
+        ActorRef deviceActor = system.actorOf(Device.props("group", "device"));
+
+        deviceActor.tell(new DeviceManager.RequestTrackDevice("group", "device"), probe.getRef());
+        probe.expectMsgClass(DeviceManager.DeviceRegistered.class);
+        Assert.assertEquals(deviceActor, probe.getLastSender());
+    }
+
+    @Test
+    public void testIgnoreWrongRegistrationRequests() {
+        TestKit probe = new TestKit(system);
+        ActorRef deviceActor = system.actorOf(Device.props("group", "device"));
+
+        deviceActor.tell(new DeviceManager.RequestTrackDevice("wrongGroup", "device"), probe.getRef());
+        probe.expectNoMsg();
```

Comment:
`expectNoMsg` should almost always be used with an explicit short (200.millis) because the default `remaining` time (3.seconds) is too long and slows down the tests.